### PR TITLE
SDEV-3122 - annotate_positional_variants - match all hgvs notations

### PR DESCRIPTION
**Pull request by dustinbleile**
https://www.bcgsc.ca/jira/browse/SDEV-3122

For structural variants matching to all hgvs notations, not just the preferred description.

### CODE REVIEW

File: ipr/annotate.py [hunk 1]
```diff
@@ -300,44 +300,50 @@ def annotate_positional_variants(
...
-            elif 'gene1' in row and 'gene2' in row:
-                problem_genes.add(row['gene1'])
-                problem_genes.add(row['gene2'])
-            else:
-                raise err
-        except HTTPError as err:
-            errors += 1
-            logger.error(f'failed to match positional variants ({variant}): {err}')
+        for var_key in VARIANT_KEYS:
+            if var_key not in row or not row[var_key]:
```
Comment (mattdoug604) [inline ipr/annotate.py:317]: `if not row.get(var_key, "")` is a little more concise and would be consistent with other checks in this function

**Review by mattdoug604 — APPROVED** [review]
Only a minor comment from me.

File: ipr/annotate.py (hunk 1 above)
Comment (dustinbleile) [reply ipr/annotate.py]: Went with this style suggestion

File: ipr/annotate.py [hunk 2]
```diff
@@ -300,44 +300,50 @@ def annotate_positional_variants(
...
-                problem_genes.add(row['gene2'])
-            else:
-                raise err
-        except HTTPError as err:
-            errors += 1
-            logger.error(f'failed to match positional variants ({variant}): {err}')
+        for var_key in VARIANT_KEYS:
+            variant = row.get(var_key, '')
+            if not variant:
+                continue
```
Comment (dustinbleile) [inline ipr/annotate.py:319]: Hi @creisle,
Do you see any issues or more elegant ways of trying this?

**Comment by elewis2** [comment]
If a variant has data for more than one of the four keys, is the desired output a new row for each of them?

**Review by elewis2 — APPROVED** [review]

**Comment by dustinbleile** [comment]
> If a variant has data for more than one of the four keys, is the desired output a new row for each of them?
Redundant rows returned would be expected with these changes.  Most hgvsProtein and hgvsCds representations should match the same variants.

I'm not sure if this is worth checking for or avoiding a this point in the code.
Maybe there would be some efficiency or memory usage gains?

**Review by RaphaelPletz — APPROVED** [review]

**Pull request merged**